subxt/src/backend/unstable/follow_stream_driver.rs

```diff
@@ -275,8 +274,26 @@ impl<Hash: BlockHash> Shared<Hash> {
                     }
                 }
 
-                // New finalized block, so clear the cache of older block events.
-                shared.block_events_from_last_finalized.clear();
+                // The blocks reported by the finalized event should not be reported again
+                // by the initialized event. The pruned blocks are not of interest.
+                let to_remove: HashSet<Hash> = finalized_ev
+                    .finalized_block_hashes
+                    .iter()
+                    .chain(finalized_ev.pruned_block_hashes.iter())
+                    .map(|h| h.hash())
+                    .collect();
+
+                shared
+                    .block_events_for_new_subscriptions
+                    .retain(|ev| match ev {
+                        FollowEvent::NewBlock(new_block_ev) => {
```

Comment:
I don't follow why `NewBlock` and `BestBlockChanged` are the only events that causes events to be removed from the `block_events`? 

Can you explain :pray:?

Reply:
or the other events removed in some other scenario?